### ISSUE TITLE
kubectl rolling-update does not accept stdin

### ISSUE THREAD

**Issue by JeanMertz** [GoogleCloudPlatform/kubernetes#8874]
```
% cat myapp.rc.json | kubectl rolling-update myapp -f -
Error: - does not specify a valid ReplicationController
```

```
% cat myapp.rc.json | kubectl update myapp -f -
replicationcontrollers/myapp
```

```
% kubectl help rolling-update

Examples:
// Update pods of frontend-v1 using JSON data passed into stdin.
$ cat frontend-v2.json | kubectl rolling-update frontend-v1 -f -
```

**Comment by thockin** [comment]
Are you sure that myapp.rc.json is valid?  Can you add --validate to this?

**Comment by bgrant0607** [comment]
It does take stdin:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/rollingupdate.go#L157

as does update:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/update.go#L99
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/resource/builder.go#L95

I think the problem is this check:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/rollingupdate.go#L172
```
		newRc, ok = obj.(*api.ReplicationController)
		if !ok {
...
			return cmdutil.UsageError(cmd, "%s does not specify a valid ReplicationController", filename)
		}
```
I think the builder generates a list when reading from a stream, unless Flatten is called, as it is here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/update.go#L100
but not here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/rollingupdate.go#L158

Probably a 1-line fix.

**Comment by bgrant0607** [comment]
cc @jlowdermilk

**Comment by jlowdermilk** [comment]
What version of kubectl are you running? I just tried to repro this from head and could not. I'm was pretty sure this was fixed by #8679.

**Comment by bgrant0607** [comment]
Believed to be working in 1.0. Please reopen if not the case.
